pkg/pool-utils/test/foundry/RecoveryMode.t.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.7.0;
+
+import "forge-std/Test.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/Math.sol";
+
+import "../../contracts/RecoveryMode.sol";
+import "../../contracts/test/MockRecoveryMode.sol";
+
+contract RecoveryModeTest is Test {
+    using FixedPoint for uint256;
+
+    uint16 private constant _TOTAL_SUPPLY = type(uint16).max;
+    uint256 private constant _MAX_TOKENS = 50;
+
+    MockRecoveryMode private _mock;
+
+    function setUp() external {
+        _mock = new MockRecoveryMode(address(0));
+    }
+
+    function testComputeProportionalAmountsOut(uint112[] memory balances, uint16 bptAmountOut) public {
+      vm.assume(balances.length <= _MAX_TOKENS);
+
+      uint256[] memory amounts = new uint256[](balances.length);
+      for (uint256 i = 0; i < balances.length; i++) {
+        amounts[i] = balances[i];
+      }
```

Comment:
```suggestion
    function testComputeProportionalAmountsOut(uint256[] memory balances, uint16 bptAmountOut) public {
      vm.assume(balances.length <= _MAX_TOKENS);

      for (uint256 i = 0; i < balances.length; i++) {
        balances[i] = bound(balances[i], 1e6, type(uint112).max);
      }
```

I think that we should probably be using bound in these cases, especially when dealing with arrays.

Reply:
Yes, that's cleaner! Didn't realize I could use bound anywhere (not just at the top); still getting the hang of this.